providers/os/id/hostname/hostname.go

```diff
@@ -65,5 +66,37 @@ func Hostname(conn shared.Connection, pf *inventory.Platform) (string, bool) {
 		}
 	}
 
+	// Fallback for windows systems to using registry for static analysis
+	if pf.IsFamily(inventory.FAMILY_WINDOWS) && conn.Capabilities().Has(shared.Capability_FileSearch) {
+		fi, err := conn.FileInfo(registry.SystemRegPath)
+		if err != nil {
+			log.Debug().Err(err).Msg("could not find SYSTEM registry file, cannot perform hostname lookup")
+			return "", false
+		}
+
+		rh := registry.NewRegistryHandler()
+		defer func() {
+			err := rh.UnloadSubkeys()
+			if err != nil {
+				log.Debug().Err(err).Msg("could not unload registry subkeys")
+			}
+		}()
+		err = rh.LoadSubkey(registry.System, fi.Path)
+		if err != nil {
+			log.Debug().Err(err).Msg("could not load SYSTEM registry key file")
+			return "", false
+		}
+		key, err := rh.GetRegistryItemValue(registry.System, "ControlSet001\\Control\\ComputerName\\ComputerName", "ComputerName")
+		if err == nil {
+			return key.Value.String, true
+		} else {
+			// we also can try ControlSet002 as a fallback
+			key, err := rh.GetRegistryItemValue(registry.System, "ControlSet002\\Control\\ComputerName\\ComputerName", "ComputerName")
+			if err == nil {
+				return key.Value.String, true
+			}
+		}
```

Comment:
no need to have this `else`

```suggestion
		}
		
		log.Debug().Err(err).Msg("unable to read windows registry, trying ControlSet002 fallback")
		key, err := rh.GetRegistryItemValue(registry.System, "ControlSet002\\Control\\ComputerName\\ComputerName", "ComputerName")
		if err == nil {
			return key.Value.String, true
		}
```